packages/opencode/src/acp/agent.ts

```
@@ -565,13 +563,14 @@ export namespace ACP {
       const sessionID = params.sessionId
       const session = this.sessionManager.get(sessionID)
       const directory = session.cwd
+      const globalCfg = await Config.get()
 
       const current = session.model
       const model = current ?? (await defaultModel(this.config, directory))
       if (!current) {
         this.sessionManager.setModel(session.id, model)
       }
-      const agent = session.modeId ?? "build"
+      const agent = session.modeId ?? globalCfg.default_agent!
```

why the "!" shouldn't this be:

globalCfg.default_agent ?? "build"?

---

oh i see ur config.

---

Hm I don't like the "!" assertion, I feel like this should behave the same way say "model" does, where we don't force the default at the config level for consistency 

So this would be 

session.modeId ?? globalCfg.default_agent ?? "build"

---

Using "build" as the default does not seem robust enough, as a user can disable it (I tried this once and it caused some problems). 

IMHO the default agent should be determined and validated (does it exist? is it enabled?) at the config level w.r.t. "fail fast" principle. It should then always be available for downstream logic.

---

that's valid